packages/admin-ui/components/pages/setup/dataSources/BigQueryProperties.tsx

```diff
@@ -4,17 +4,50 @@ import UploadOutlined from '@ant-design/icons/UploadOutlined';
 import { UploadFile } from 'antd/lib/upload/interface';
 import { ERROR_TEXTS } from '@vulcan-sql/admin-ui/utils/error';
 
-export default function BigQueryProperties() {
+const UploadCredentials = (props) => {
+  const { onChange } = props;
+
   const [fileList, setFileList] = useState<UploadFile[]>([]);
 
+  const convertFileToJSON = (file: any, callback: (value: JSON) => void) => {
+    const reader = new FileReader();
+    reader.onloadend = (_e) => {
+      const result = reader.result;
+
+      if (result) {
+        const fileContent: JSON = JSON.parse(String(result));
+        callback(fileContent);
+      }
+    };
+
+    reader.readAsText(file);
+  };
+
   const onUploadChange = (info) => {
-    const { fileList } = info;
+    const { file, fileList } = info;
     if (fileList.length) {
       const uploadFile = fileList[0];
+      convertFileToJSON(file.originFileObj, (fileContent: JSON) => {
+        onChange && onChange(fileContent);
+      });
       setFileList([uploadFile]);
     }
   };
 
+  return (
+    <Upload
+      accept=".json"
+      fileList={fileList}
+      onChange={onUploadChange}
+      onRemove={() => setFileList([])}
```

Comment:
The onRemove function also needs to update onChange to the form instance

```ts
const onRemove = () => {
  setFileList([])
  onChange && onChange(null)
}
```

Reply:
Thanks for the reminder.
I updated the code, remove will make it to the initial state.